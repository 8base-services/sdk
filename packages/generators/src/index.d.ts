import { TableSchema } from '@8base/utils';

export const REPO_BRANCH_NAME: string;

export type GeneratorsCommonConfig = {
  deep: number,
  withMeta?: boolean,
};

export type GeneratorsConfig = {
  includeColumns?: string[],
} & GeneratorsCommonConfig;

export type GeneratorsData = {
  tablesList: TableSchema,
  tableName: string,
  screenName?: string,
};

export type ScreenTable = {
  id?: string,
  tableName: string,
  screenName: string,
  routeUrl?: string,
  tableFields?: string[],
  formFields?: string[],
};


export function replaceInitialApp(fsObject: Object, constants: {
  endpoint: string,
  authClientId: string,
  authDomain: string,
  appName: string,
}): Object;

export function generateScreen(data: {
  tablesList: TableSchema[],
  screen: ScreenTable,
  rootFile?: string,
}, config: GeneratorsCommonConfig): Object;
