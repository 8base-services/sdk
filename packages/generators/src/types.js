// @flow
import type { TableSchema } from '@8base/utils';

export type GeneratorsConfig = {
  deep: number,
  includeColumns?: string[],
};

export type GeneratorsData = {
  tablesList: TableSchema,
  tableName: string,
  screenName?: string,
};

export type ScreenTable = {
  id: string,
  tableName: string,
  screenName: string,
  routeUrl: string,
  tableFields?: string[],
  formFields?: string[],
};
