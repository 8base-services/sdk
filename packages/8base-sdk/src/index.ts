export * from '@8base/apollo-client';
export * from '@8base/apollo-links';
export * from '@8base/auth';
export * from '@8base/utils';
export * from '@8base/validate';
