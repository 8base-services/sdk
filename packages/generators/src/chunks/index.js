// @flow
import ejs from 'ejs';

// $FlowIgnore
import fieldsInputsList from './fieldsInputsList.js.ejs';
// $FlowIgnore
import fieldsQueries from './fieldsQueries.js.ejs';
// $FlowIgnore
import fieldsFileInput from './fieldsFileInput.js.ejs';

export const chunks = {
  fieldsInputsList: (data?: Object) => ejs.render(fieldsInputsList, data),
  fieldsQueries: (data?: Object) => ejs.render(fieldsQueries, data),
  fieldsFileInput: (data?: Object) => ejs.render(fieldsFileInput, data),
};

