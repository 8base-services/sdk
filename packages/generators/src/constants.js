// @flow

/** Name of thi branch in repo https://github.com/8base/react-app-starter */
export const REPO_BRANCH_NAME = 'v1.1.0';
