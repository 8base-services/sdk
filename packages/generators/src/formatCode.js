// @flow
import prettier from 'prettier/standalone';
import prettierBabylon from 'prettier/parser-babylon';

export const formatCode = (code: string) => prettier.format(code, {
  printWidth: 120,
  singleQuote: true,
  trailingComma: 'es5',
  parser: 'babel',
  plugins: [prettierBabylon],
});
