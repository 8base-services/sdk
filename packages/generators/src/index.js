// @flow

export { generateCreateForm } from './generateCreateForm/createForm';
export { generateEditForm } from './generateEditForm/editForm';
export { generateDeleteForm } from './generateDeleteForm/deleteForm';
export { generateTable } from './generateTable/table';
export { generateIndex } from './generateIndex';

