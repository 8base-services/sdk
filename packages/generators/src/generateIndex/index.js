// @flow

import ejs from 'ejs';
import pluralize from 'pluralize';
import changeCase from 'change-case';
import { formatCode } from '../formatCode';

// $FlowIgnore
import index from './index.js.ejs';

export const generateIndex = ({ tableName, screenName }: *) => {
  const entityName = pluralize.singular(tableName);

  const tableGenerated = ejs.render(index, {
    changeCase,
    pluralize,
    tableName,
    entityName,
    screenName: screenName || entityName,
  });

  return formatCode(tableGenerated);
};

