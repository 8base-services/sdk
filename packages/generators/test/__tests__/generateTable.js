// @flow

import { generateTable } from '../../src';
import { tablesSchema } from '../__fixtures__/tablesSchema';


it('should generate table by the table name', () => {
  const generatedTable = generateTable(tablesSchema, 'Properties');

  expect(generatedTable).toMatchSnapshot();
});

