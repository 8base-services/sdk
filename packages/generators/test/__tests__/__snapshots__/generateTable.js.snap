// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate table by the table name 1`] = `
"import React from 'react';
import { compose } from 'recompose';
import { pathOr } from 'ramda';
import gql from 'graphql-tag';
import { TableBuilder, Dropdown, Icon, Menu, withModal } from '@8base/boost';
import { graphql } from 'react-apollo';

const BROKERS_LIST_QUERY = gql\`
  
  query DataViewerTablePropertiesContent($filter: PropertyFilter, $orderBy: [PropertyOrderBy], $after: String, $before: String, $first: Int, $last: Int, $skip: Int) { 
    tableContent: propertiesList(filter: $filter, orderBy: $orderBy, after: $after, before: $before, first: $first, last: $last, skip: $skip) {
      items {
        pictures {
          items {
            id
            fileId
            filename
            downloadUrl
            shareUrl
            meta
          }
          count
        }
        bedrooms
        title
        description
        sqFootage
        bathrooms
        garage
        pool
        listings {
          items {
            documents {
              items {
                id
                fileId
                filename
                downloadUrl
                shareUrl
                meta
              }
              count
            }
            status
            closingDate
            price
          }
          count
        }
        _description
      }
      count
    }
  }
\`;

const TABLE_COLUMNS = [
  
    {
      name: bedrooms,
      title: Bedrooms,
    },
  
    {
      name: title,
      title: Title,
    },
  
    {
      name: description,
      title: Description,
    },
  
    {
      name: sqFootage,
      title: SqFootage,
    },
  
    {
      name: bathrooms,
      title: Bathrooms,
    },
  
    {
      name: garage,
      title: Garage,
    },
  
    {
      name: pool,
      title: Pool,
    },
  
    {
      name: listings,
      title: Listings,
    },
  
  {
    name: 'edit',
    title: '',
    width: '60px',
  },
];

const enhancer = compose(
  withModal,
  graphql(PROPERTY_LIST_QUERY, { name: 'properties' })
);

const PropertyTable = enhancer(({ (properties, openModal, closeModal }) => (
  <TableBuilder
    loading={properties.loading}
    data={pathOr([], ['propertiesList', 'items'], properties)}
    columns={TABLE_COLUMNS}
    renderCell={(column, rowData) => {
      switch (column.name) {
        case 'edit':
          return (
            <Dropdown defaultOpen={false}>
              <Dropdown.Head>
                <Icon name=\\"Dots\\" size=\\"sm\\" color=\\"LIGHT_GRAY2\\" />
              </Dropdown.Head>
              <Dropdown.Body pin=\\"right\\">
                {({ closeDropdown }) => (
                  <Menu>
                    <Menu.Item
                      onClick={() => {
                        openModal(PropertyDeleteDialog.id, { id: rowData.id });
                        closeDropdown();
                      }}
                    >
                      Delete
                    </Menu.Item>
                  </Menu>
                )}
              </Dropdown.Body>
            </Dropdown>
          );
        default:
          return <div>{pathOr('', column.name.split('.'), rowData)}</div>;
      }
    }}
    action=\\"Create Property\\"
    onActionClick={() => openModal(PropertyCreateDialog.id)}
  />
));

export { PropertyTable };
"
`;
